src/NBitcoin/ChainedHeader.cs

```diff
@@ -286,15 +286,19 @@ public ChainedHeader FindAncestorOrSelf(ChainedHeader chainedHeader)
         }
 
         /// <summary>
-        /// Finds the ancestor of this entry in the chain that matches the block hash given.
+        /// Finds the ancestor of this entry in the chain that matches the block hash, or optionally, the block height given.
         /// </summary>
         /// <param name="blockHash">The block hash to search for.</param>
+        /// <param name="height">Optional height to search for.</param>
         /// <returns>The ancestor of this chain that matches the block hash.</returns>
-        public ChainedHeader FindAncestorOrSelf(uint256 blockHash)
+        public ChainedHeader FindAncestorOrSelf(uint256 blockHash, int height = -1)
```

Comment:
The default value of zero would be even better

Reply:
Then you have to do the extra check in the while loop, its fine like I recon.

Reply:
Why would you need any extra check?

Reply:
Because zero is also the genesis block.

Reply:
Yes, so why would you need any extra checks?

Reply:
Okay, it should return null, you are right. Returns comment should be updated to mention what should happen when the element is not found.

Reply:
Ah should we return null in case the max limit was reached?

Reply:
LOL, that is a correct question and answer is probably yes. We should totally change the comment to know what we are doing

Reply:
Okay, so now, as we have the new implementation, let me again ask my original question – would it not be better if the default value was zero on the index?